contracts/accessories/Flashloan.vy

```diff
@@ -0,0 +1,41 @@
+from ethereum.ercs import IERC20
+
+interface IERC3156FlashBorrower:
+    def onFlashLoan(
+        initiator: address,
+        token: IERC20,
+        amount: uint256,
+        fee: uint256,
+        data: Bytes[65535],
+    ) -> bytes32: nonpayable
+
+
+implements: IERC3156FlashBorrower
+
+from .. import IAccessory
+implements: IAccessory
+
+
+@pure
+@external
+def getMethodIds() -> DynArray[bytes4, 100]:
+    return [0x23e30c8b]
+
+
+@external
+def onFlashLoan(
+    initiator: address,
+    token: IERC20,
+    amount: uint256,
+    fee: uint256,
+    data: Bytes[65535],
+) -> bytes32:
+    # NOTE: Ensure that appropiate amount of allowance is given to caller
+    if staticcall token.allowance(tx.origin, msg.sender) < amount + fee:
+        extcall token.approve(msg.sender, amount + fee)
```

Comment:
Similarly to [this security disclosure](https://github.com/yearn/yearn-security/blob/master/disclosures/2021-06-29.md), this function would allow arbitrarily approving `msg.sender` so that they could snatch away the balance here. This function should therefore be restricted only to a trusted EOA context to avoid the problem:
```suggestion
    # NOTE: Only trusted context allowed
    assert tx.origin == self, "Flashloan:!authorized"

    # NOTE: Ensure that appropriate amount of allowance is given to caller
    if staticcall token.allowance(tx.origin, msg.sender) < amount + fee:
        extcall token.approve(msg.sender, amount + fee)
```